spiped/dispatch.c

```diff
@@ -61,8 +61,9 @@ callback_resolve(void * cookie, struct sock_addr ** sas)
 	}
 
 	/* Wait a while before resolving again. */
-	if (events_timer_register_double(callback_resolveagain,
-	    A, A->rtime) == NULL)
+	A->dnstimer_cookie = events_timer_register_double(callback_resolveagain,
+	    A, A->rtime);
+	if (A->dnstimer_cookie == NULL)
```

Comment:
Our code style would write this as:
```
if ((A->dnstimer_cookie = events_blah(blah)) == NULL)
```
I personally prefer your version, but I know that it won't be accepted as-is.  Also, having a consistent style is more important than any one particular stylistic preference.  :)